scrapy/linkextractors/lxmlhtml.py

```diff
@@ -20,8 +18,9 @@
 
 _collect_string_content = etree.XPath("string()")
 
+
 def _nons(tag):
-    if isinstance(tag, basestring):
+    if isinstance(tag, six.string_types):
```

Comment:
Test coverage shows that this condition is always True. 

@redapple do you recall when can elem.tag be something other than string? Can we remove pass-through from _nons?